cupy/__init__.py

```diff
@@ -686,6 +686,24 @@ def isscalar(element):
 # -----------------------------------------------------------------------------
 from cupy.core import size  # NOQA
 
+
+def ndim(a):
+    """Returns the number of dimensions of an array.
+
+    Args:
+        a (array-like): If it is not already an `cupy.ndarray`, a conversion
+            via :func:`numpy.asarray` is attempted.
+
+    Returns:
+        (int): The number of dimensions in `a`.
+
+    """
+    try:
+        return a.ndim
+    except AttributeError:
+        return numpy.asarray(a).ndim
```

Comment:
How about this:
```suggestion
        return numpy.ndim(a)
```
This fallback will cover Python scalars and sequences as well.

Reply:
Yeah, I think that makes sense and covers most use cases.

Either approach still has issues with some types of input, though. For example, if the user were to input a list of large CuPy arrays as in the following:

```Python
np.ndim([cp.ones((1024, 1024))] * 4)
``` 

This takes several seconds to run!

I got a bit confused about what the desired behavior was in these edge cases and it stalled there.